polaris-react/src/utilities/use-event-listener.ts

```diff
@@ -0,0 +1,31 @@
+import {RefObject, useEffect, useRef} from 'react';
+
+type ElementType = Window | Document | HTMLElement | RefObject<HTMLElement>;
+
+export function useEventListener(
+  event: string,
+  handler: (event: Event) => void,
+  element: ElementType = window,
+  options?: AddEventListenerOptions,
+) {
+  const handlerRef = useRef(handler);
+  const optionsRef = useRef(options);
+
+  useEffect(() => {
+    handlerRef.current = handler;
+    optionsRef.current = options;
+  }, [handler, options]);
```

Comment:
```suggestion
  const handlerRef = useRef(handler);
  const optionsRef = useRef(options);

  useEffect(() => {
    handlerRef.current = handler;
  }, [handler]);

  useEffect(() => {
    optionsRef.current = options;
  }, [options]);
```

Thoughts on separating `handlerRef` and `optionsRef` assignments. If `handler` was memoized and `options` wasn't we would end up reassigning `handler` every render.

Reply:
I think that's overkill personally. I'll leave it up to you in the follow up